test/models/namespace_group_link_test.rb

```diff
@@ -65,4 +65,17 @@ def setup
       I18n.t('activerecord.errors.models.namespace_group_link.attributes.namespace_type.inclusion')
     )
   end
+
+  test '#scope for_namespace_and_ancestors returns the correct collection' do
+    group_group_link = namespace_group_links(:namespace_group_link2)
+
+    group = group_group_link.namespace
+
+    shared_with_group_links = group.shared_with_group_links
+    namespace_group_links = NamespaceGroupLink.for_namespace_and_ancestors(group)
+
+    shared_with_group_links.each do |shared_with_group_link|
```

Comment:
**nitpick (blocking):** Can you add a test to ensure that the counts are the same. That way we know that both collections are the same.

Reply:
Updated test to check that the collections and counts are the same in 07d46704b0aab5511c116e8688c9ae977526cd96